### PR TITLE
MULE-11032: Enable online dependency resolution for artifact function…

### PR DESCRIPTION
**Pull request by gsfernandes**
…al testcase

### CODE REVIEW

File: tests/runner/src/main/java/org/mule/test/runner/api/RepositorySystemFactory.java [hunk 1]
```diff
@@ -103,4 +128,6 @@ public void serviceCreationFailed(Class<?> type, Class<?> impl, Throwable except
     return locator.getService(RepositorySystem.class);
   }
 
+
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/api/RepositorySystemFactory.java]: Revert this change

File: tests/runner/src/main/java/org/mule/test/runner/api/RepositorySystemFactory.java [hunk 2]
```diff
@@ -53,17 +54,41 @@
    * @param workspaceLocationResolver {@link WorkspaceLocationResolver} to resolve artifactId's {@link Path}s from workspace. Not
    *        {@code null}.
    */
-  public static DependencyResolver newLocalDependencyResolver(List<URL> classPath,
-                                                              WorkspaceLocationResolver workspaceLocationResolver,
-                                                              File mavenLocalRepositoryLocation) {
-    DefaultRepositorySystemSession session = newSession();
+  public static DependencyResolver newOfflineDependencyResolver(List<URL> classPath,
```
Comment (pablokraan) [inline tests/runner/src/main/java/org/mule/test/runner/api/RepositorySystemFactory.java]: No unit test for this?

**Comment by gsfernandes** [comment]
https://github.com/mulesoft/mule/pull/3677
